common/shader_util.hpp

```diff
@@ -17,78 +19,82 @@ static void GLAPIENTRY
 					[[maybe_unused]] GLsizei length,
 					[[maybe_unused]] const GLchar* message,
 					[[maybe_unused]] const void* userParam )
-	{
+{
 #ifndef NDEBUG
-    if (severity == GL_DEBUG_SEVERITY_NOTIFICATION)
-    {
+    if (severity == GL_DEBUG_SEVERITY_NOTIFICATION) {
         return; // Don't show notification-severity messages.
     }
-	fprintf( stderr, "GL CALLBACK: %s type = 0x%x, severity = 0x%x, message = %s\n",
-			( type == GL_DEBUG_TYPE_ERROR ? "** GL ERROR **" : "" ),
-				type, severity, message );
+    std::cerr << "GL CALLBACK: " << (type == GL_DEBUG_TYPE_ERROR ? "** GL ERROR **" : "")
+              << " type = 0x" << std::hex << type << std::dec
+              << ", severity = 0x" << std::hex << severity << std::dec
+              << ", message = " << message
+              << std::endl;
+    ILLIXR::abort();
```

Comment:
We check more explicitly using the `severity` interface above. Continuing to use `severity` over `type` might be more robust.

Either we can:
- Return early for `GL_DEBUG_SEVERITY_LOW` and `GL_DEBUG_SEVERITY_MEDIUM` like for `GL_DEBUG_SEVERITY_NOTIFICATION`, or
- Only return early for `GL_DEBUG_SEVERITY_NOTIFICATION`, and guard the `abort` for `GL_DEBUG_SEVERITY_LOW` and `GL_DEBUG_SEVERITY_MEDIUM` so that we get the print out for low, medium, and high, but only crash when it's high.

Reply:
Option 2 sounds great, but let's crash with medium too. This is just to err on the side of caution.